include/sympack/Environment.hpp

```diff
@@ -137,7 +151,34 @@ namespace symPACK{
 #include "sympack/LogFile.hpp"
 
 
-
+#define CUBLAS_ERROR_CHECK(s)                                                                       \
+{                                                                                                   \
+    cublasStatus_t err;                                                                             \
+    if ((err = (s)) != CUBLAS_STATUS_SUCCESS)                                                       \
+    {                                                                                               \
+        std::cout << "cuBLAS Error " << err << " at " << __FILE__ << ":" << __LINE__ << "\n";       \
+        exit(1);                                                                                    \
+    }                                                                                               \
+}                                                                                                   \
+
+#define CUDA_ERROR_CHECK(s)                                                                         \
+{                                                                                                   \
+    cudaError_t error = s;                                                                          \
+    if (error != cudaSuccess) {                                                                     \
+        std::cout << "CUDA Error " << error << " at " << __FILE__ << ":" << __LINE__ << "\n";       \
+        std::cout << cudaGetErrorString(error) << "\n";                                             \
+        exit(1);                                                                                    \
+    }                                                                                               \
+}   
+
+#define CUSOLVER_ERROR_CHECK(s)                                                                          \
+{                                                                                                        \
+    cusolverStatus_t error = s;                                                                          \
+    if (error != CUSOLVER_STATUS_SUCCESS) {                                                              \
+        std::cout << "cuSOLVER Error " << error << " at " << __FILE__ << ":" << __LINE__ << "\n";        \
+        exit(1);                                                                                         \
+    }                                                                                                    \
+}                                                                                                        \
```

Comment:
For block-statement macros like this, it's best-practice to wrap the block definition inside a `do { ... } while(0)` statement. This idiom ensures non-surprising behavior with invocations of the form:
```
if (something()) CUDA_ERROR_CHECK(some_cuda_call());
else             CUDA_ERROR_CHECK(some_other_cuda_call());
```
which would otherwise generate a parse error as the macros are currently defined.

```suggestion
#define CUBLAS_ERROR_CHECK(s)                                                                       \
do {                                                                                                   \
    cublasStatus_t err;                                                                             \
    if ((err = (s)) != CUBLAS_STATUS_SUCCESS)                                                       \
    {                                                                                               \
        std::cout << "cuBLAS Error " << err << " at " << __FILE__ << ":" << __LINE__ << "\n";       \
        exit(1);                                                                                    \
    }                                                                                               \
} while (0)

#define CUDA_ERROR_CHECK(s)                                                                         \
do {                                                                                                   \
    cudaError_t error = s;                                                                          \
    if (error != cudaSuccess) {                                                                     \
        std::cout << "CUDA Error " << error << " at " << __FILE__ << ":" << __LINE__ << "\n";       \
        std::cout << cudaGetErrorString(error) << "\n";                                             \
        exit(1);                                                                                    \
    }                                                                                               \
} while (0)

#define CUSOLVER_ERROR_CHECK(s)                                                                          \
do {                                                                                                        \
    cusolverStatus_t error = s;                                                                          \
    if (error != CUSOLVER_STATUS_SUCCESS) {                                                              \
        std::cout << "cuSOLVER Error " << error << " at " << __FILE__ << ":" << __LINE__ << "\n";        \
        exit(1);                                                                                         \
    }                                                                                                    \
} while (0)
```